python/hail/context.py

```diff
@@ -365,7 +365,13 @@ def read(self, path, sites_only=False, samples_only=False):
         :rtype: :class:`.VariantDataset`
         """
 
-        jvds = self._jhc.readAll(jindexed_seq_args(path), sites_only, samples_only)
+        is_generic_genotype = self._jhc.readAllMetadata(jindexed_seq_args(path)).isGenericGenotype()
```

Comment:
I'm not totally sold on this -- reading metadata can be slow. Is there any way to check without reading metadata twice?